File: web/src/lib/api/types/beta-guide.ts
export type BetaGuideStep = {
  key: string;
  label: string;
  hint: string | null;
};

export type BetaGuide = {
  current_phase: string;
  what_to_test: string[];
  expected_path: BetaGuideStep[];
  known_limitations: string[];
  feedback_instructions: string;
};

export type BetaAccess = {
  status: "pending" | "approved" | "blocked";
  gate_enabled: boolean;
  can_use_mvp: boolean;
  safe_message: string | null;
};

File: app/schemas/beta_guide.py
"""Read-only beta tester guide (Phase AI.97)."""

from __future__ import annotations

from pydantic import BaseModel, Field


class BetaGuideStep(BaseModel):
    key: str
    label: str
    hint: str | None = None


class BetaGuideResponse(BaseModel):
    current_phase: str = "closed_beta_mvp"
    what_to_test: list[str] = Field(default_factory=list)
    expected_path: list[BetaGuideStep] = Field(default_factory=list)
    known_limitations: list[str] = Field(default_factory=list)
    feedback_instructions: str = ""
